Tests/Common/SimulatedBasicsStation.cs

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoRaWan.Tests.Common
+{
+
+    using System;
+    using System.Text.Json;
+    using LoRaWan.Tests.Simulation.Models;
+    using Websocket.Client;
+    using System.IO;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using System.Runtime.InteropServices;
+    using LoRaWan.NetworkServer;
+    using LoRaTools.LoRaMessage;
+    using LoRaTools.Utils;
+    using System.Net.WebSockets;
+    using Xunit;
+    using System.Linq;
+
+    public sealed class SimulatedBasicsStation : IDisposable
+    {
+        private readonly string stationEUI;
+        private WebsocketClient DataWebsocketClient { get; set; }
+        private Uri LnsUri { get; set; }
+
+        private readonly HashSet<Func<string, bool>> subscribers = new HashSet<Func<string, bool>>();
+
+        internal void SubscribeOnce(Func<string, bool> value)
+        {
+            this.subscribers.Add(value);
+        }
+
+
+        public SimulatedBasicsStation(string stationEUI, Uri lnsUri)
+        {
+            this.stationEUI = stationEUI;
+            this.LnsUri = lnsUri;
+        }
+
+        public async Task StartAsync()
+        {
+            try
+            {
+                var routerReceivedMessage = new List<string>();
+                using var routerWebsocketClient = new WebsocketClient(new Uri(LnsUri, "router-info"))
+                {
+                    ReconnectTimeout = TimeSpan.FromSeconds(5)
+                };
+                routerWebsocketClient.ReconnectionHappened.Subscribe(info =>
+                {
+                    Console.WriteLine("Reconnection happened, type: " + info.Type);
+                });
+                routerWebsocketClient.MessageReceived.Subscribe(msg =>
+                {
+                    routerReceivedMessage.Add(msg.Text);
+                    Console.WriteLine("Message received: " + msg);
+                });
+                await routerWebsocketClient.Start();
+                using var ms = new MemoryStream();
+                using var writer = new Utf8JsonWriter(ms);
+
+                await routerWebsocketClient.SendInstant(JsonSerializer.Serialize(new
+                {
+                    router = this.stationEUI
+                }));
+
+                await Task.Delay(5000);
```

Comment:
It would be good if we could figure out a way to wait for the response instead of having the static timeout here.

Reply:
SHould be done now